### PR TITLE
ci: add extra checks

### PR DESCRIPTION
**Pull request by sreimers**

### CODE REVIEW

**Comment by mbattista** [comment]
How do you think about even more checks with different environments?

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
Offers mac and windows, which could make future releases simpler (https://github.com/baresip/baresip/issues/1074)

**Comment by alfredh** [comment]
this looks great!

should we merge it to master for the 1.1.0 release ?

File: .github/workflows/extra.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    - name: make HAVE_PTHREAD= baresip
+      run: |
+        make V=1 CCACHE= EXTRA_CFLAGS=-Werror HAVE_PTHREAD=
+        make clean
+
+    - name: make RELEASE=1 baresip
+      run: |
+        make V=1 CCACHE= EXTRA_CFLAGS=-Werror RELEASE=1
+        make clean
```
Comment (cspiel1) [inline .github/workflows/extra.yml:61]: no newline at end of file

Comment (cspiel1) [reply .github/workflows/extra.yml:61]: yes, looks great! Will not hurt in 1.1.0.

**Pull request merged**